lib/bulk_sql_delete.rb

```diff
@@ -0,0 +1,62 @@
+class BulkSqlDelete
+  attr_accessor :connection
+
+  def initialize(connection = ActiveRecord::Base.connection)
+    self.connection = connection
+  end
+
+  def delete_in_batches(sql)
+    perform_and_log(sql) do
+      unless Rails.env.test?
+        connection.begin_db_transaction
+      end
+      result = connection.exec_delete(sql)
+      connection.commit_db_transaction unless Rails.env.test?
```

Comment:
We should use connection pooling here, so that we return the connection to the pool at the end. This is usually done inside the requests or when we use the model but here we're outside both:

```ruby
ActiveRecord::Base.connection_pool.with_connection do |connection|
end
```

the connection will be returned to the pool at the end of the block

Reply:
a question: why we skip the transaction statements in tests?

Reply:
It will mess up your test db bc the test db cleaners rely on transactions to rollback data after the specs are run and if you start committing data in the middle of a spec you will get leftover data thats not cleaned up.